### PR TITLE
hw-6 v-0.0.1

### PR DESCRIPTION
**Pull request by odykyi**

### CODE REVIEW

File: api/user/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+    const msg = {
+      to: email,
+      from: process.env.EMAIL,
+      subject: "Verification",
+      text: `http://localhost:${process.env.PORT}/verify/${verificationToken}`,
+      html: `<p>Please, <a href=http://localhost:${process.env.PORT}/auth/verify/${verificationToken}>click</a> to verify your email</p>`,
+    };
+
+    sgMail.send(msg);
```
Comment (odykyi) [inline api/user/user.controller.js:79]: await

File: api/user/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    const avatarName = Date.now() + ".png";
+
+    const avatar = await catAvatar.create(email);
+
+    await fsPromise.writeFile(avatarPath + "/" + avatarName, avatar);
+
+    return `http://localhost:${process.env.PORT}/images/${avatarName}`;
+  };
+
+  static sendVerificationEmail = (email, verificationToken) => {
```
Comment (odykyi) [inline api/user/user.controller.js]: async

File: api/user/user.controller.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+      const avatarURL = await this.avatarGenerator(req.body.email);
+
+      const passwordHash = await bcrypt.hash(
+        req.body.password,
+        +process.env.BCRYPT_SALT_ROUNDS
+      );
+
+      const verificationToken = uuidv4();
+
+      this.sendVerificationEmail(req.body.email, verificationToken);
```
Comment (odykyi) [inline api/user/user.controller.js]: await

File: api/user/user.controller.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+      });
+    } catch (err) {
+      next(err);
+    }
+  };
+
+  static avatarGenerator = async (email) => {
+    const catAvatar = Avatar.catBuilder(256);
+
+    const avatarPath = path.join(__dirname, "../public/images");
```
Comment (odykyi) [inline api/user/user.controller.js:57]: move to config

File: api/user/user.controller.js [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+    const avatarName = Date.now() + ".png";
+
+    const avatar = await catAvatar.create(email);
+
+    await fsPromise.writeFile(avatarPath + "/" + avatarName, avatar);
+
+    return `http://localhost:${process.env.PORT}/images/${avatarName}`;
```
Comment (odykyi) [inline api/user/user.controller.js]: move to config

File: api/user/user.controller.js [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+  };
+
+  static sendVerificationEmail = (email, verificationToken) => {
+    sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+    const msg = {
+      to: email,
+      from: process.env.EMAIL,
+      subject: "Verification",
+      text: `http://localhost:${process.env.PORT}/verify/${verificationToken}`,
```
Comment (odykyi) [inline api/user/user.controller.js]: move to config

File: api/user/user.controller.js [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+    } catch (err) {
+      next(err);
+    }
+  };
+
+  static updateAvatar = async (req, res, next) => {
+    try {
+      await this.minifyImage(req.file.path);
+
+      const path = `http://localhost:${process.env.PORT}/images/${req.file.filename}`;
```
Comment (odykyi) [inline api/user/user.controller.js]: move to config

File: api/user/user.controller.js [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+      res.status(200).send({ avatarURL: path });
+    } catch (err) {
+      next(err);
+    }
+  };
+
+  static minifyImage = async (tmpFilePath) => {
+    await imagemin([tmpFilePath], {
+      destination: "api/public/images",
```
Comment (odykyi) [inline api/user/user.controller.js:269]: move to config

File: api/user/user.router.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
+const express = require("express");
+const path = require("path");
+const userController = require("./user.controller");
+const multer = require("multer");
+
+const storage = multer.diskStorage({
+  destination: "api/tmp",
```
Comment (odykyi) [inline api/user/user.router.js:8]: move to config

File: api/app.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    console.log(err);
+    process.exit(1);
+  }
+};
+
+const startServer = async () => {
+  await dbConnect();
+
+  app.listen(process.env.PORT, () =>
+    console.log(`App listening at http://localhost:${process.env.PORT}`)
```
Comment (odykyi) [inline api/app.js]: move to config

**Review by alexbondify — APPROVED** [review]

**Pull request merged**